lib/matplotlib/backend_bases.py

```diff
@@ -2791,6 +2785,15 @@ def mouse_move(self, event):
             except (ValueError, OverflowError):
                 pass
             else:
+                artists = event.inaxes.hitlist(event)
+                if event.inaxes.patch in artists:
+                    artists.remove(event.inaxes.patch)
+                artists.sort(key=lambda x: x.zorder)
+                if artists:
```

Comment:
Something seems odd about checking if `artists` is empty *after* sorting the list. I know (provided `artists` is a list) it won't error out, but still seems to me that the sort should go ahead and move within the `if` block.

Also, +1 for removing ye olden style DSU in favor of `sort(key=)`.